grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java

```diff
@@ -221,4 +228,22 @@ private static void attachSingleGeneratedRelation(Representation currentHal, Con
             throw new RuntimeException("Unrecognized base type of " + type);
         }
     }
+
+    static void generateConceptState(Representation resource, Concept concept){
+
+        if (concept.isInstance()) {
+            Instance instance = concept.asInstance();
+            resource.withProperty(ID_PROPERTY, instance.getId().getValue())
```

Comment:
I think it's confusing that in the case of an instance the ID property is filled with a concept's ID